jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataGrid.java

```diff
@@ -0,0 +1,214 @@
+package com.epam.jdi.light.elements.complex.table;
+
+import com.epam.jdi.light.asserts.generic.table.IDataGridAssert;
+import com.epam.jdi.light.elements.base.UIBaseElement;
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.complex.ISetup;
+import com.epam.jdi.light.elements.complex.WebList;
+import com.epam.jdi.light.elements.interfaces.base.HasInit;
+import com.epam.jdi.light.elements.interfaces.base.HasValue;
+import com.epam.jdi.light.elements.interfaces.composite.PageObject;
+import com.epam.jdi.tools.LinqUtils;
+import com.epam.jdi.tools.Safe;
+import com.epam.jdi.tools.func.JFunc1;
+import org.openqa.selenium.WebElement;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.epam.jdi.light.common.Exceptions.exception;
+import static com.epam.jdi.light.elements.pageobjects.annotations.WebAnnotationsUtil.getElementName;
+import static com.epam.jdi.light.settings.JDISettings.ELEMENT;
+import static com.epam.jdi.tools.ReflectionUtils.*;
+import static com.epam.jdi.tools.StringUtils.setPrimitiveField;
+import static com.epam.jdi.tools.StringUtils.splitCamelCase;
+import static java.util.Arrays.asList;
+
+/**
+ * Created by Roman Iovlev on 26.09.2019
+ * Email: roman.iovlev.jdi@gmail.com; Skype: roman.iovlev
+ */
+public class DataGrid<L extends PageObject, D> extends UIBaseElement<IDataGridAssert<D, IDataGrid<L, D>,?>>
+        implements ISetup, IGrid<D>, IDataGrid<L, D>, HasInit {
+    protected Safe<Grid> grid = new Safe<>(Grid::new);
+    public Grid grid() {
+        return grid.get();
+    }
+    public void init() {
+        Grid element = grid.get();
+        element.init();
+        grid = new Safe<>(() -> element);
+    }
+    protected Class<D> dataClass = null;
```

Comment:
Codacy found an issue: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6397461)